### ISSUE TITLE
System.Net.Sockets.SocketException (0x80004005): Address already in use

### ISSUE THREAD

**Issue by MarLoe** [richardschneider/net-mdns#6]
I am getting this exception when using the latest NuGet version (0.5.0) i Microsoft Visual Studio for Mac.
If I download this repo, and reference the Mdns project directly, it works perfectly.

```
Unhandled Exception:
System.Net.Sockets.SocketException (0x80004005): Address already in use
  at System.Net.Sockets.Socket.SetSocketOption (System.Net.Sockets.SocketOptionLevel optionLevel, System.Net.Sockets.SocketOptionName optionName, System.Object optionValue) [0x000ee] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/System/System.Net.Sockets/Socket.cs:2461 
  at Makaretu.Dns.MulticastService.FindNetworkInterfaces (System.Object state) [0x0013c] in <bad354b0b01c4ec3b13cbe9f835c3cc2>:0 
  at System.Threading.Timer+Scheduler.TimerCB (System.Object o) [0x00007] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Threading/Timer.cs:327 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00015] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:1279 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:856 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:1211 
[ERROR] FATAL UNHANDLED EXCEPTION: System.Net.Sockets.SocketException (0x80004005): Address already in use
  at System.Net.Sockets.Socket.SetSocketOption (System.Net.Sockets.SocketOptionLevel optionLevel, System.Net.Sockets.SocketOptionName optionName, System.Object optionValue) [0x000ee] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/System/System.Net.Sockets/Socket.cs:2461 
  at Makaretu.Dns.MulticastService.FindNetworkInterfaces (System.Object state) [0x0013c] in <bad354b0b01c4ec3b13cbe9f835c3cc2>:0 
  at System.Threading.Timer+Scheduler.TimerCB (System.Object o) [0x00007] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Threading/Timer.cs:327 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00015] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:1279 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:856 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:1211 
```

**Comment by richardschneider** [comment]
Thanks for raising the issue.  As you probably guessed, I forgot to release the latest software to NuGet.  Please try using v0.6.0.  If it works for you, then just close the issue,

**Comment by richardschneider** [comment]
BTW, No testing has been done on MacOS, because I don't have one.  So please keep me informed on any issues.

**Comment by MarLoe** [comment]
Hey!
Created a PR for MacOS compatibility.

**Comment by richardschneider** [comment]
Available in v0.8,0
